compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -117,13 +117,8 @@ TR::Register *
 OMR::Z::TreeEvaluator::bconstEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    {
    TR::Register * tempReg = node->setRegister(cg->allocateRegister());
-   if (node->getOpCodeValue() == TR::buconst)
-      generateLoad32BitConstant(cg, node, node->getByte() & 0xFF, tempReg, true);
-   else
-      {
-      TR_ASSERT(1, "Should not have bconst node in the final il-tree!\n");
-      generateLoad32BitConstant(cg, node, node->getByte(), tempReg, true);
-      }
+   TR_ASSERT(1, "Should not have bconst node in the final il-tree!\n");
```

Comment:
@fjeremic is this assert even valid? this should be removed shouldn't it (in the context of OMR)

Reply:
@fjeremic Ping

Reply:
This assert seems bogus. `bconst` trees should be valid in the OMR context. I suspect this may be some Java-ism leaking over. This evaluator looks correct. We should remove the assert.

Reply:
I have deleted the assert in c56d3c5.